src/app/foundations/chart/chart-demo.component.ts

```diff
@@ -0,0 +1,194 @@
+    chartType = 'bar';
+    chartData = 'simple';
+    showZoom = true;
+    showLegend = true;
+    chartOptions = "";
+    myChart;
+    resizeObserver: ResizeObserver;
+
+    constructor(sectionService: SectionService) {
+        super(sectionService);
+        this.buildChartOptions();
+    }
+
+    ngAfterViewInit(): void {
+        const theme = {...(this.chartTheme === 'light' ? hcChartLight : hcChartDark)}; // json is loaded as a module spreading it to reconstruct it as an object.
+        this.chartContainer.nativeElement.style.background = this.chartTheme === 'light' ? "#fff": "#222";
+        this.myChart = echarts.init(this.chartContainer.nativeElement, theme);
+        this.resizeObserver = new ResizeObserver(() => this.myChart.resize());
+        this.resizeObserver.observe(this.chartContainer.nativeElement);
+        this.updateChart();
+    }
+
+    ngOnDestroy(): void {
+        echarts.dispose(this.chartContainer.nativeElement);
+        this.resizeObserver && this.resizeObserver.disconnect();
+    }
+
+    buildChartOptions(): Record<string, unknown> {
+        const data = exampleData[this.chartData];
+        const options = {
+            legend: this.showLegend ? {} : undefined,
+            xAxis: {
+              type: 'category',
+              data: data.labels,
+            },
+            yAxis: {
+              type: 'value'
+            },
+            series: data.series.map(it => ({
+                type: this.chartType,
+                label: this.chartData === 'simple' ? {
+                    show: true,
+                    rotate: 0,
+                    align: 'center',
+                    verticalAlign: 'middle',
+                    position: 'top',
+                    distance: 15,
+                    fontSize: 12,
+                    rich: {
+                        name: {}
+                    }
+                } : undefined,
+                ...it,
+            })),
+            tooltip: {
+                trigger: 'axis',
+                axisPointer: {
+                    type: 'shadow'
+                }
+            },
+            dataZoom: this.showZoom ? [
+                {
+                    type: 'inside',
+                }, {}
+            ] : undefined,
+
+        };
+        this.chartOptions = "```json\n" + JSON.stringify(options, null, 2) + "\n```\n";
+        return options;
+    }
+
+    updateChart():void {
+        if(!this.chartContainer) {
+            console.warn("Chart container not present!");
+            return;
+        }
+
+        this.myChart.clear();
+        this.myChart.setOption(this.buildChartOptions());
+    }
+
+    changeChartType(chartType: SelectChangeEvent): void {
+        this.chartType = chartType.value;
+        this.updateChart();
+    }
+    
+    changeChartTheme(chartTheme: SelectChangeEvent): void {
+        this.chartTheme = chartTheme.value;
+        echarts.dispose(this.chartContainer.nativeElement);
+        this.ngAfterViewInit();
```

Comment:
Calling `this.ngAfterViewInit()` seems like bad practice. It would be better to create a `initChart()` and call it here and inside `ngAfterViewInit()`.